src/components/common/Builder/FormBuilder/types.ts

```diff
@@ -0,0 +1,35 @@
+import type { FC } from "react";
+import type {
+    SubmitHandler,
+    UseFormStateReturn,
+    ControllerFieldState,
+    ControllerRenderProps,
+} from "react-hook-form";
+import type { PropertyMeta } from "../../../../services/hubspot/types";
+import type { Layout } from "../types";
+
+export type Meta = Record<PropertyMeta["name"], PropertyMeta>;
+
+export type Config = {
+    structure: Layout;
+    metaMap: Meta;
+};
+
+export type FieldsMap = Record<PropertyMeta["fieldType"], FC>;
+
+export type FormBuilderProps = {
+    config: Config;
+    fieldsMap: FieldsMap;
+    onSubmit: SubmitHandler<Record<string, unknown>>;
```

Comment:
Why is the type `unknown`? This is our code so we should be able to strongly type this.

Reply:
Yes, this is our code, but it is based on the metadata of the custom field from the API. And we don't know what the possible values might be, so we're not sure. And we handle the values separately in distinct implementations of the custom fields `./src/components/fields/*`

Reply:
I made a slight adjustment to the type like this: `Record<PropertyMeta["name"], unknown>`